app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt

```diff
@@ -0,0 +1,197 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabtray
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import androidx.core.view.isVisible
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.SimpleItemAnimator
+import kotlinx.android.extensions.LayoutContainer
+import kotlinx.android.synthetic.main.component_tab_tray.view.*
+import kotlinx.android.synthetic.main.tab_tray_list_item.view.*
+import mozilla.components.feature.media.state.MediaState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.components.FenixSnackbar
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.components.ui.SelectableListItemView
+import org.mozilla.fenix.components.ui.SelectionHolder
+import org.mozilla.fenix.components.ui.SelectionInteractor
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.increaseTapArea
+import org.mozilla.fenix.library.LibraryPageView
+import org.mozilla.fenix.utils.allowUndo
+
+interface TabTrayViewInteractor : SelectionInteractor<Tab> {
+    fun closeButtonTapped(tab: Tab)
+    fun onPauseMediaClicked()
+    fun onPlayMediaClicked()
+    fun privateModeButtonTapped()
+    fun normalModeButtonTapped()
+    fun closeAllTabsTapped()
+    fun newTabTapped()
+}
+
+class TabItemViewHolder(
```

Comment:
Can we pull TabItemViewHolder, and TabTrayAdapter out of this file into `TabTrayAdapter.kt`?